File: frontend-admin/src/types/product/product-option.ts
export interface VariantOptionValue {
  productOptionValue: ProductOptionValue;
  id?: number;
  variantId: number;
  optionId: number;
  priceModifier: number;
}
export interface ProductOptionValue {
  id?: number;
  value: string;
  displayValue: string;
  sortOrder?: number;
}

export interface ProductOption {
  id?: number;
  name: string;
  displayName: string;
  isRequired: boolean;
  sortOrder?: number;
  values: ProductOptionValue[];
}

export type OptionType = "SELECT" | "TEXT" | "NUMBER";

File: platform-ecom-backend/product/src/main/java/com/ecom/product/entity/ProductOption.java
package com.ecom.product.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "product_options")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString(exclude = "values")
public class ProductOption {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(name = "display_name", nullable = false, length = 100)
    private String displayName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @Builder.Default
    private Boolean isRequired = false;

    @Column(name = "sort_order")
    @Builder.Default
    private Integer sortOrder = 0;

    @OneToMany(mappedBy = "option", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sortOrder ASC")
    @Builder.Default
    private List<ProductOptionValue> values = new ArrayList<>();
}
